lib/wave_gen/wave_gen.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdlib.h>
+
+#include <math.h>
+#ifndef M_PI
+  #define M_PI 3.14159265358979323846
+#endif
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wave_gen, CONFIG_WAVE_GEN_LIB_LOG_LEVEL);
+
+#include <wave_gen.h>
+
+/**
+ * @brief Generates a pseudo-random number between -1 and 1.
+ */
+static double generate_pseudo_random(void)
+{
+	return (double)rand() / ((double)RAND_MAX / 2.0) - 1.0;
```

Comment:
`srand()` could be called at some point to avoid generating the same number sequence each time

Reply:
Thanks for pointing this out

Reply:
Still, I wonder if this part should be actually done by the library.

Maybe it could be better to inform in docs that `rand()` function is called to generate the random noise and it's up to the library user to call `srand`. For simulated data, it may be better to actually generate "the same noise" in some use-cases (e.g. when we are testing an algorithm we may want to test on signals with predefined noise).

Reply:
btw. In current implementation the `srand(k_cycle_get_32())` is called in `sensor_sim` driver initialization.

Reply:
Do you think it would be better to e.g. provide some API to initialize the wave_gen library (the API would be actually a wrapper on srand call) ?

Reply:
Yes, an initialization API sounds like a good idea